--- a/drivers/input/touchscreen/mediatek/hxchipset_hx83102p/Kconfig
+++ b/drivers/input/touchscreen/mediatek/hxchipset_hx83102p/Kconfig
@@ -16,7 +16,7 @@ config TOUCHSCREEN_HIMAX_COMMON
 
 choice
 	prompt "Himax touch panel IC series"
-	depends on TOUCHSCREEN_HIMAX_COMMON
+	depends on TOUCHSCREEN_HIMAX_COMMON || TOUCHSCREEN_HIMAX_HX83102P
 	default TOUCHSCREEN_HIMAX_INCELL
 
 config TOUCHSCREEN_HIMAX_ONCELL
@@ -25,7 +25,7 @@ config TOUCHSCREEN_HIMAX_ONCELL
 
 config TOUCHSCREEN_HIMAX_INCELL
 	bool "Himax in-cell function"
-	depends on TOUCHSCREEN_HIMAX_COMMON
+	depends on TOUCHSCREEN_HIMAX_COMMON || TOUCHSCREEN_HIMAX_HX83102P
 	help
 	  Switch for on-cell or in-cell
 	  Say Y here if you have Himax in-cell touch panel